cookbooks/inner_core_convection/inner_core_convection.cc

```diff
@@ -60,6 +103,64 @@ namespace aspect
 {
   namespace MaterialModel
   {
+    template <int dim>
+    InnerCore<dim>::InnerCore ()
+      :
+      resistance_to_phase_change (1),
+      hydrostatic_pressure_profile (1)
+    {}
+
+    template <int dim>
+    double
+    InnerCore<dim>::
+    phase_function (const Point<dim> &position,
+                    const double temperature) const
+    {
+      // We need to convert the depth to pressure,
+      // keeping in mind that for this model, the pressure is only the dynamic pressure,
+      // so we have to get the hydrostatic pressure explicitly as an input.
+      const double radius = this->get_geometry_model().maximal_depth() - this->get_geometry_model().depth(position);
+      const double pressure_deviation = hydrostatic_pressure(radius) - hydrostatic_pressure(transition_radius);
+
+      double depth_deviation = transition_radius - radius;
+      if (depth_deviation != 0.0)
```

Comment:
In floating point arithmetic, this condition is always true. You will want to compare against something that is small.